src/components/NavBar/Profile/index.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { useContext, useState } from 'react';
+import { useHistory } from 'react-router-dom';
+import AccountCircle from '@material-ui/icons/AccountCircle';
+import Button from '@material-ui/core/Button';
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import { Auth0Lock } from 'auth0-lock';
+import SmallLogo from '../../../utils/assets/small-logo.png';
+import { store } from '../../../state/store';
+
+import { ProfileCollapse, ProfileIcon } from './style';
+
+const Profile = () => {
+  const [anchorEl, setAnchorEl] = useState(null);
+  const globalState = useContext(store);
+  const { dispatch } = globalState;
+  const myStorage = window.sessionStorage;
+  const history = useHistory();
+
+  const clientId = process.env.REACT_APP_CLIENT_ID || '';
+  const domain = process.env.REACT_APP_DOMAIN || '';
+
+  const options = {
+    theme: {
+      logo: SmallLogo,
+      primaryColor: 'var( --greenblue)',
+      languageDictionary: {
+        title: 'Log in',
+      },
+    },
+  };
+  const lock = new Auth0Lock(clientId, domain, options);
+  let profile = null;
+
+  lock.on('authenticated', function login(authResult) {
+    lock.getUserInfo(authResult.accessToken, function getUserInfo(error, profileResult) {
+      if (error) {
+        return;
+      }
+      profile = profileResult;
+      console.log(profile);
+      myStorage.setItem('loggedUser', JSON.stringify(profile));
+      dispatch({ type: 'login', payload: profile });
+    });
+  });
+
```

Comment:
When testing this component, I'll try to mock all the functions related to the Auth0 dependencies. I think it can help you for the different scenarios you may encounter for this component.